metal/v1/model_device.go

```diff
@@ -1,1680 +1,71 @@
+ *
+ * API version: 1.0.0
+ * Contact: support@equinixmetal.com
+ * Generated by: Swagger Codegen (https://github.com/swagger-api/swagger-codegen.git)
+ */
+package swagger
 
 import (
-	"encoding/json"
 	"time"
 )
 
-// checks if the Device type satisfies the MappedNullable interface at compile time
-var _ MappedNullable = &Device{}
-
-// Device struct for Device
 type Device struct {
-	AlwaysPxe           *bool                  `json:"always_pxe,omitempty"`
-	BillingCycle        *string                `json:"billing_cycle,omitempty"`
-	BondingMode         *int32                 `json:"bonding_mode,omitempty"`
-	CreatedAt           *time.Time             `json:"created_at,omitempty"`
-	CreatedBy           *DeviceCreatedBy       `json:"created_by,omitempty"`
-	Customdata          map[string]interface{} `json:"customdata,omitempty"`
-	Description         *string                `json:"description,omitempty"`
-	Facility            *Facility              `json:"facility,omitempty"`
-	HardwareReservation *HardwareReservation   `json:"hardware_reservation,omitempty"`
-	Hostname            *string                `json:"hostname,omitempty"`
-	Href                *string                `json:"href,omitempty"`
-	Id                  *string                `json:"id,omitempty"`
-	ImageUrl            *string                `json:"image_url,omitempty"`
-	IpAddresses         []IPAssignment         `json:"ip_addresses,omitempty"`
-	IpxeScriptUrl       *string                `json:"ipxe_script_url,omitempty"`
-	Iqn                 *string                `json:"iqn,omitempty"`
+	AlwaysPxe           bool                  `json:"always_pxe,omitempty"`
+	BillingCycle        string                `json:"billing_cycle,omitempty"`
+	BondingMode         int32                 `json:"bonding_mode,omitempty"`
+	CreatedAt           time.Time             `json:"created_at,omitempty"`
+	CreatedBy           *AllOfDeviceCreatedBy `json:"created_by,omitempty"`
+	Customdata          ModelMap              `json:"customdata,omitempty"`
```

Comment:
openapi-generator definitely gives a nicer result for this attribute and the one above it